server/build.gradle.kts

```diff
@@ -60,25 +63,28 @@ tasks {
     }
     
     val startProduction by creating(JavaExec::class) {
+        group = "application"
         dependsOn(makeRunnable)
-        classpath = files(configurations.default, runnableDir.resolve("software-challenge-server.jar"))
-        main = "sc.server.Application"
-        workingDir = runnableDir
+        classpath = jar.get().outputs.files
         jvmArgs = listOf("-Dfile.encoding=UTF-8", "-Dlogback.configurationFile=../../logback-production.xml", "-Djava.security.egd=file:/dev/./urandom", "-XX:MaxGCPauseMillis=100", "-XX:GCPauseIntervalMillis=2050", "-XX:+UseConcMarkSweepGC", "-XX:+CMSParallelRemarkEnabled", "-XX:+UseCMSInitiatingOccupancyOnly", "-XX:CMSInitiatingOccupancyFraction=70", "-XX:+ScavengeBeforeFullGC", "-XX:+CMSScavengeBeforeRemark")
     }
     
     val dockerImage by creating(Exec::class) {
+        group = "application"
         dependsOn(makeRunnable)
+        workingDir = projectDir.resolve("configuration")
 		doFirst {
 			val tag = Runtime.getRuntime().exec(arrayOf("git", "rev-parse", "--short", "--verify", "HEAD")).inputStream.reader().readText().trim()
-			val relativeRunnable = runnableDir.relativeTo(project.projectDir)
+			val relativeRunnable = runnableDir.relativeTo(workingDir)
 			commandLine("docker", "build", "--no-cache", "-t", "swc_game-server:latest", "-t", "swc_game-server:$tag", "--build-arg", "game_server_dir=$relativeRunnable", ".")
```

Comment:
This won't work with the `relativeRunnable` path `/server/build/runnable` being outside the working dir `/server/configuration`. All resources for the container have to be under the working directory which is specified by the last argument of the `docker build` command (currently `.`). You can copy `/server/build/runnable` into the working dir or give docker another working directory and change the paths to the other resources in `Dockerfile`. I would prefer the first option.

```
> Task :server:dockerImage FAILED
Sending build context to Docker daemon  11.78kB

Step 1/9 : FROM openjdk:8-jre
 ---> 39f48bacf78c
Step 2/9 : ARG game_server_dir
 ---> Running in 4527c39a980f
Removing intermediate container 4527c39a980f
 ---> de29cfdb46e1
Step 3/9 : ADD $game_server_dir /app
ADD failed: Forbidden path outside the build context: ../build/runnable ()

FAILURE: Build failed with an exception.
```

Reply:
please check the new version

Reply:
works with my additional changes :+1: